### PR TITLE
Cdp manager

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

**Review by hexonaut — COMMENTED** [review]
I like the general strategy. This will probably require a new audit. Some comments about the details.

File: src/CharterManager.sol [hunk 1]
```diff
@@ -110,13 +105,8 @@ contract CharterManager {
     }
 }
 
-contract CharterManagerImp {
-    // --- Proxy Storage ---
-    bytes32 slot0; // avoid collision with proxy's implementation field
-    mapping (address => uint256) public wards;
-
+contract CharterManagerImp is ProxyStorage, SubCdpManager {
```
[the remark this answers is not in the document]
Comment (talbaneth) [reply src/CharterManager.sol:108]: Yeah the main reason I went for inheritance is that the sub manager will be fully replicated in crop join. And once that is inherited we have to inherit the proxy storage as well in order for it to be first.

**Comment by talbaneth** [comment]
btw the code is compiling if you remove the other contracts (thue unit tests file can stay).

**Comment by talbaneth** [comment]
Something else to consider/discuss is whether to have the main manager contract configureable (using `file()`). That would make it easier to change (avoiding an upgrade). Guess it depends on how soon and frequent the main manager would be replaced.

**Comment by talbaneth** [comment]
Decided to go with cdp-registry solution.

**Pull request closed without merging**